app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ActionController::Base
+  def create
+    # user_post = Post.find(params[:post_id])
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @new_comment = @post.comments.new(user_id: @user.id, post_id: @post_id, text: comment_params['text'])
+    # new_comment = Comment.new(author: user, post: user_post, text: comment_params)
+
+    if @new_comment.save
+      # @new_comment.count_comments(@post.id)
+      Comment.count_comments(@post.id)
+      flash[:success] = 'Your comment has been added!'
+      redirect_to user_post_path(@user, @post)
+    else
+      flash.now[:error] = 'Comment could not be added'
+      # render user_post_path(@user, @post)
+    end
+  end
```

Comment:
- [ ] To keep your code clean and professional please remove the commented code.